code/game/gamemodes/objective.dm

```diff
@@ -364,11 +408,11 @@
 	return 1
 
 
-/datum/objective/martyr
+/datum/objective/escape_obj/martyr

```

Comment:
In general I don't think you want lings having to die because if a ling needs to die it's going to go out with the maximum number of kills it can.
Also I'm not sure the absorb objective is compatable with gibbing. It should be, but I'm not sure if it is.